dockers/docker-fpm-frr/frr.conf.j2

```diff
@@ -79,6 +79,62 @@ route-map FROM_BGP_SPEAKER_V4 permit 10
 !
 route-map TO_BGP_SPEAKER_V4 deny 10
 !
+{% block vnet_bgp_instance %}
+{% if VNET is defined %}
+! Vnet BGP instance
+{% for vnet_name, vnet_metadata in VNET.iteritems() %}
+router bgp {{ DEVICE_METADATA['localhost']['bgp_asn'] }} vrf {{ vnet_name }}
+  no bgp default ipv4-unicast
+{% for (name, prefix) in LOOPBACK_INTERFACE %}
+{% if prefix | ipv4 and name == 'Loopback0' %}
+  bgp router-id {{ prefix | ip }}
+{% endif %}
+{% endfor %}
+{# got interfaces that belong this vnet #}
+{% set interfaces_in_vnet = [] %}
+{% for ((name, prefix), metadata) in INTERFACE.iteritems() %}
+{% if metadata.has_key("vnet_name") and metadata["vnet_name"] == vnet_name %}
+{% do interfaces_in_vnet.append( prefix | ip ) %}
+{% endif %}
+{% endfor %}
+{% for neighbor_addr, bgp_session in BGP_NEIGHBOR.iteritems() %}
+{% if bgp_session.has_key("local_addr") and bgp_session["local_addr"] in interfaces_in_vnet %}
+{% if bgp_session['asn'] | int != 0 %}
+  neighbor {{ neighbor_addr }} remote-as {{ bgp_session['asn'] }}
+  neighbor {{ neighbor_addr }} description {{ bgp_session['name'] }}
+{# set the bgp neighbor timers if they have not default values #}
+{% if     (bgp_session['keepalive'] is defined and bgp_session['keepalive'] | int != 60)
+      or  (bgp_session['holdtime'] is defined  and bgp_session['holdtime']  | int != 180) %}
+  neighbor {{ neighbor_addr }} timers {{ bgp_session['keepalive'] }} {{ bgp_session['holdtime'] }}
+{% endif %}
+{% if bgp_session.has_key('admin_status') and bgp_session['admin_status'] == 'down' or not bgp_session.has_key('admin_status') and DEVICE_METADATA['localhost'].has_key('default_bgp_status') and DEVICE_METADATA['localhost']['default_bgp_status'] == 'down' %}
+  neighbor {{ neighbor_addr }} shutdown
+{% endif %}
+{% if neighbor_addr | ipv4 %}
+  address-family ipv4
+{% if DEVICE_METADATA['localhost']['type'] == 'ToRRouter' %}
+    neighbor {{ neighbor_addr }} allowas-in 1
```

Comment:
Can you describe the circumstances, setup and requirements that need the allowas-in so that I can better understand?

Reply:
it is needed as the tor asn is re-used for different clusters.